File: exercises/beecrowd/1179/1179.ts
// https://judge.beecrowd.com/pt/problems/view/1179
import * as fs from "fs";

function main(): void {
  const lines: number[] = fs
    .readFileSync("/dev/stdin", "utf8")
    .trim()
    .split("\n")
    .map(Number);

  let lineIndex: number = 0;

  let evenNumbers: number[] = [];
  let oddNumbers: number[] = [];

  for (let i: number = 0; i < 15; i++) {
    const num: number = lines[lineIndex++];

    if (num % 2 === 0) {
      evenNumbers.push(num);
      if (evenNumbers.length === 5) {
        for (let j: number = 0; j < 5; j++) {
          console.log(`par[${j}] = ${evenNumbers[j]}`);
        }
        evenNumbers = [];
      }
    } else {
      oddNumbers.push(num);
      if (oddNumbers.length === 5) {
        for (let j: number = 0; j < 5; j++) {
          console.log(`impar[${j}] = ${oddNumbers[j]}`);
        }
        oddNumbers = [];
      }
    }
  }

  for (let i: number = 0; i < oddNumbers.length; i++) {
    console.log(`impar[${i}] = ${oddNumbers[i]}`);
  }

  for (let i: number = 0; i < evenNumbers.length; i++) {
    console.log(`par[${i}] = ${evenNumbers[i]}`);
  }
}

main();

File: exercises/beecrowd/1179/1179.go
// https://judge.beecrowd.com/pt/problems/view/1179
package main

import "fmt"

func main() {
	var evenNumbers []int
	var oddNumbers []int

	for i := 0; i < 15; i++ {
		var num int
		fmt.Scan(&num)

		if num%2 == 0 {
			evenNumbers = append(evenNumbers, num)
			if len(evenNumbers) == 5 {
				for j := 0; j < 5; j++ {
					fmt.Printf("par[%d] = %d\n", j, evenNumbers[j])
				}
				evenNumbers = nil
			}
		} else {
			oddNumbers = append(oddNumbers, num)
			if len(oddNumbers) == 5 {
				for j := 0; j < 5; j++ {
					fmt.Printf("impar[%d] = %d\n", j, oddNumbers[j])
				}
				oddNumbers = nil
			}
		}
	}

	for i, num := range oddNumbers {
		fmt.Printf("impar[%d] = %d\n", i, num)
	}

	for i, num := range evenNumbers {
		fmt.Printf("par[%d] = %d\n", i, num)
	}
}
